jujugui/static/gui/src/app/components/deployment/input/input.js

```diff
@@ -0,0 +1,111 @@
+/*
+This file is part of the Juju GUI, which lets users view and manage Juju
+environments within a graphical interface (https://launchpad.net/juju-gui).
+Copyright (C) 2016 Canonical Ltd.
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU Affero General Public License version 3, as published by
+the Free Software Foundation.
+
+This program is distributed in the hope that it will be useful, but WITHOUT
+ANY WARRANTY; without even the implied warranties of MERCHANTABILITY,
+SATISFACTORY QUALITY, or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Affero
+General Public License for more details.
+
+You should have received a copy of the GNU Affero General Public License along
+with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+'use strict';
+
+YUI.add('deployment-input', function() {
+
+  juju.components.DeploymentInput = React.createClass({
+
+    propTypes: {
+      disabled: React.PropTypes.bool,
+      label: React.PropTypes.string.isRequired,
+      placeholder: React.PropTypes.string.isRequired,
+      required: React.PropTypes.bool,
+      validate: React.PropTypes.array,
+      value: React.PropTypes.string,
+    },
+
+
+    getDefaultProps: () => {
+      return {
+        required: false
+      };
+    },
+
+    getInitialState: function() {
+      return {errors: null};
+    },
+
+    /**
+      Validate the field value.
+
+      @method validate
+    */
+    validate: function() {
+      var value = this.refs.field.value;
```

Comment:
You could use your new `getValue` function to get the value so all values are accessed the same way.